File: one-intent-hyperliquid-wallet-complete-reviewed-2026-07-29.3/apps/android/app/src/main/java/jp/offlinewallet/ReviewContract.kt
package jp.offlinewallet

/**
 * Pure offline review contract. READY only means local final-review UI; it
 * never authorizes signing, broadcasting, a deep link, or a network request.
 */
data class ReviewContractV1(
    val sourceUtterance: String,
    val normalizedInterpretation: String,
    val materialAmbiguities: List<String>,
    val userConfirmed: Boolean = false,
) {
    enum class ReasonCode {
        INVALID_CONTRACT,
        BLOCKED_AMBIGUITIES,
        BLOCKED_EXPLICIT_CONFIRMATION_REQUIRED,
        READY_FOR_LOCAL_FINAL_REVIEW,
    }

    val reasonCode: ReasonCode
        get() = when {
            sourceUtterance.isBlank() ||
                normalizedInterpretation.isBlank() ||
                materialAmbiguities.any { it.isBlank() } ||
                materialAmbiguities.size != materialAmbiguities.distinct().size ||
                (userConfirmed && materialAmbiguities.isNotEmpty()) -> ReasonCode.INVALID_CONTRACT
            materialAmbiguities.isNotEmpty() -> ReasonCode.BLOCKED_AMBIGUITIES
            !userConfirmed -> ReasonCode.BLOCKED_EXPLICIT_CONFIRMATION_REQUIRED
            else -> ReasonCode.READY_FOR_LOCAL_FINAL_REVIEW
        }

    val isValid: Boolean
        get() = reasonCode != ReasonCode.INVALID_CONTRACT

    val primaryActionEnabled: Boolean
        get() = reasonCode == ReasonCode.READY_FOR_LOCAL_FINAL_REVIEW
}

File: one-intent-hyperliquid-wallet-complete-reviewed-2026-07-29.3/apps/ios/Sources/OfflineWalletContract/ReviewContract.swift
import Foundation

public struct ReviewContract: Sendable, Equatable {
    public let sourceUtterance: String
    public let normalizedInterpretation: String
    public let materialAmbiguities: [String]
    public let userConfirmed: Bool

    public init(sourceUtterance: String, normalizedInterpretation: String, materialAmbiguities: [String], userConfirmed: Bool = false) {
        self.sourceUtterance = sourceUtterance
        self.normalizedInterpretation = normalizedInterpretation
        self.materialAmbiguities = materialAmbiguities
        self.userConfirmed = userConfirmed
    }

    public enum ReasonCode: String, Sendable {
        case invalidContract = "INVALID_CONTRACT"
        case blockedAmbiguities = "BLOCKED_AMBIGUITIES"
        case blockedExplicitConfirmationRequired = "BLOCKED_EXPLICIT_CONFIRMATION_REQUIRED"
        case readyForLocalFinalReview = "READY_FOR_LOCAL_FINAL_REVIEW"
    }

    public var reasonCode: ReasonCode {
        if sourceUtterance.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty ||
            normalizedInterpretation.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty ||
            materialAmbiguities.contains(where: { $0.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty }) ||
            materialAmbiguities.count != Set(materialAmbiguities).count ||
            (userConfirmed && !materialAmbiguities.isEmpty) {
            return .invalidContract
        }
        if !materialAmbiguities.isEmpty { return .blockedAmbiguities }
        if !userConfirmed { return .blockedExplicitConfirmationRequired }
        return .readyForLocalFinalReview
    }

    public var isValid: Bool { reasonCode != .invalidContract }

    public var primaryActionEnabled: Bool {
        reasonCode == .readyForLocalFinalReview
    }
}
